GitCommands/Git/GitModule.cs

```diff
@@ -2103,6 +2103,16 @@ public string CommitCmd(bool amend, bool signOff = false, string author = "", bo
                 command += " --author=\"" + author + "\"";
             }                
 
+            if (gpgSign)
+            {
+                command += " -S";
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(gpgKeyId))
```

Comment:
I find it confusing - if we are not signing then we should not be signing.
If we are supplying `gpgKeyId` we should have `gpgSign=true`.

Reply:
This can be little confusing because of the two variable evaluation needed.
`gpgSign` is a bool that comes from `SelectedIndex == 1` evaluation, that's true only if Default Key is selected.
Otherwise, if text is present in `gpgKeyId` means that there's a specific key, if it's empty no sign should be done.
`gpgSign` name can be misleading in this case.

It's possible to change this part to an enum with three values or more complex implementation, but GitModule is more complex to be refactored instead of only add this easy feature.

Reply:
FormCommit line:1017
if you change `gpgSignCommitToolStripComboBox.SelectedIndex == 1` to `gpgSignCommitToolStripComboBox.SelectedIndex > 0` it will have `gpgSign=true`
Then lines:2106-2114 in GitModule can be refactored and simplified.

Reply:
True, I'm changing it.